Entities and repos for Juror Summons Replies

Matt-Smith-CGI opened the pull request:
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines

### Change description ###

-Added relevant classes for Juror Summons Reply
-Added generator tags to relevant fields
-Added relevant repos for Juror Summons

MaffYouuuu committed:
Added repos 3 repos from backend

MaffYouuuu committed:
-Removed PoliceCheckRepository.java
-Removed Querydsl from PoolCommentRepository

MaffYouuuu committed:
-Removed JurorPoolRepository.java

MaffYouuuu committed:
-Added relevant classes for Juror Summons Reply
-Added generator tags to relevant fields
-Added relevant repos for Juror Summons

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/AbstractJurorResponse.java:
```diff
@@ -0,0 +1,136 @@
+@Table(name = "juror_response", schema = "juror_mod")
+@Getter
+@Setter
+
```
Space not needed

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/AbstractJurorResponse.java:
```diff
@@ -0,0 +1,136 @@
+@Getter
+@Setter
+
+public class AbstractJurorResponse extends Address implements Serializable {
```
Should be an abstract class as we do not want this to be instantiated by itself

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/AbstractJurorResponse.java:
```diff
@@ -0,0 +1,136 @@
+    private String jurorNumber;
+
+    @Column(name = "date_received")
+    private LocalDate dateReceived;
```
Needs date generator

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/AbstractJurorResponse.java:
```diff
@@ -0,0 +1,136 @@
+    private String lastName;
+
+    @Column(name = "processing_status")
+    private String processingStatus;
```
Needs default value

Ben-Edwards-cgi replied:
Should be: 

  @Column(name = "processing_status")
    @Enumerated(EnumType.STRING)
    @FixedValueGenerator("ProcessingStatus.TODO")
    private ProcessingStatus processingStatus = ProcessingStatus.TODO;

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/AbstractJurorResponse.java:
```diff
@@ -0,0 +1,136 @@
+    private String email;
+
+    @Column(name = "residency")
+    private Boolean residency;
```
All columns need default can be done using @FixedValueGenerator("your default")

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/Address.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Column(name = "address_line_1")
+    @NotBlank
+    @RegexGenerator(regex = "[0-9]{0,9} [A-Z]{5,15} (Road|Lane|Gate|Close|Avenue|Street|Way|Drive|Gardens|Crescent|"
```
Regex can be moved to Constants class

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/Address.java:
```diff
@@ -0,0 +1,58 @@
+    private String addressLine1;
+
+    @Column(name = "address_line_2")
+    @RegexGenerator(regex = "(Apartment|Suite|Room|Floor|Box) Number [0-9]{1,3}")
```
Regex can be moved to Constants class

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,118 @@
+public class DigitalResponse extends AbstractJurorResponse {
+
+    @Column(name = "residency")
+    @Builder.Default
```
Needs default using @FixedValueGenerator.
(Same for all fields with no generator)

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/PaperResponse.java:
```diff
@@ -0,0 +1,38 @@
+public class PaperResponse extends AbstractJurorResponse {
+
+    @Column(name = "mental_health_capacity")
+    private Boolean mentalHealthCapacity;
```
Needs defaults

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/ReasonableAdjustments.java:
```diff
@@ -0,0 +1,25 @@
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "t_reasonable_adjustments", schema = "juror_mod")
```
Any constant table (annotation with t_) is not needed

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/ReplyType.java:
```diff
@@ -0,0 +1,26 @@
+import java.io.Serializable;
+
+@Entity
+@Table(name = "t_reply_type", schema = "juror_mod")
```
Class not needed as we can use Id's instead

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/repository/JurorDigitalResponseRepository.java:
```diff
@@ -0,0 +1,15 @@
+
+@Repository
+public interface JurorDigitalResponseRepository extends JurorResponseRepository<DigitalResponse> {
+    DigitalResponse findByJurorNumberAndReplyType(String jurorNumber, ReplyType replyType);
+
+    default DigitalResponse findByJurorNumber(String jurorNumber) {
+        return findByJurorNumberAndReplyType(jurorNumber, new ReplyType("Digital", null));
+    }
```
We only use repositories to save as such no methods are needed

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/repository/JurorPaperResponseRepository.java:
```diff
@@ -0,0 +1,15 @@
+
+@Repository
+public interface JurorPaperResponseRepository extends JurorResponseRepository<PaperResponse> {
+    PaperResponse findByJurorNumberAndReplyType(String jurorNumber, ReplyType replyType);
+
+    default PaperResponse findByJurorNumber(String jurorNumber) {
+        return findByJurorNumberAndReplyType(jurorNumber, new ReplyType("Paper", null));
+    }
```
Methods not needed

MaffYouuuu committed:
Addressing comments on MR
-Removed ReplyType.java
-Made amendments relating to MR comments

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private String deferralDate;
+
+    @Column(name = "excusal_reason")
+    @FixedValueGenerator("d")
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+
+    @Column(name = "deferral_reason")
+    @FixedValueGenerator("false")
+    private String deferralReason;
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private Boolean mentalHealthAct = Boolean.FALSE;
+
+    @Column(name = "residency_detail")
+    @FixedValueGenerator("true")
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private String residencyDetail;
+
+    @Column(name = "mental_health_act_details")
+    @FixedValueGenerator("false")
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private String mentalHealthActDetails;
+
+    @Column(name = "bail_details")
+    @FixedValueGenerator("false")
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private String bailDetails;
+
+    @Column(name = "convictions_details")
+    @FixedValueGenerator("false")
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private String deferralReason;
+
+    @Column(name = "deferral_date")
+    @LocalDateGenerator(
+            minInclusive = @DateFilter(mode = DateFilter.Mode.PLUS, value = 4, unit = ChronoUnit.MONTHS),
+            maxExclusive = @DateFilter(mode = DateFilter.Mode.PLUS, value = 2, unit = ChronoUnit.WEEKS)
+    )
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private Integer version;
+
+    @Column(name = "thirdparty_fname")
+    @FirstNameGenerator
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+    private String thirdPartyFName;
+
+    @Column(name = "thirdparty_lname")
+    @LastNameGenerator
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+
+    @Column(name = "thirdparty_other_reason")
+//    @Pattern(regexp = NO_PIPES_REGEX)
+    @FixedValueGenerator("ill")
```
Should default to null

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/DigitalResponse.java:
```diff
@@ -0,0 +1,127 @@
+
+    @Column(name = "juror_email_details")
+    @Builder.Default
+    @EmailGenerator
```
Should default to true

Ben-Edwards-cgi commented on src/main/java/uk/gov/hmcts/juror/support/sql/entity/jurorresponse/ReasonableAdjustments.java:
```diff
@@ -0,0 +1,23 @@
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+public class ReasonableAdjustments {
+    @Column(name = "code")
+    @Id
+    private String code;
+
+    @Column(name = "description")
+    private String description;
+}
```
Not needed can use id

MaffYouuuu committed:
Addressing MR comments

MaffYouuuu committed:
Addressing MR comments
-Removed irrelevant classes;
--ReasonableAdjustments.java
-Created ProcessingStatus.java
-Amendments to DigitalResponse.java and AbstractJurorResponse.java

Ben-Edwards-cgi committed:
Merge branch 'master' into jurorSummonsReplies

# Conflicts:
#	src/main/java/uk/gov/hmcts/juror/support/sql/Constants.java

Ben-Edwards-cgi committed:
Minor updates to align code sets

Matt-Smith-CGI reviewed (commented):
reviewed
